src/crypto/crypto_util.h

```diff
@@ -367,25 +367,23 @@ class CryptoJob : public AsyncWrap, public ThreadPoolWork {
     v8::Local<v8::Value> args[2];
     {
       node::errors::TryCatchScope try_catch(env);
-      v8::Maybe<bool> ret = ptr->ToResult(&args[0], &args[1]);
-      if (!ret.IsJust()) {
+      if (ptr->ToResult(&args[0], &args[1]).IsNothing()) {
         CHECK(try_catch.HasCaught());
+        CHECK(try_catch.CanContinue());
         exception = try_catch.Exception();
-      } else if (!ret.FromJust()) {
+        ptr->MakeCallback(env->ondone_string(), 1, &exception);
```

Comment:
This moves the call into the TryCatchScope – that's probably not intentional, right? Exceptions from `MakeCallback` should end up being treated as uncaught exceptions

Reply:
Ah right, good catch! :-)